Integrations/Gmail/Gmail.py

```diff
@@ -1028,22 +1330,297 @@ def remove_filter(user_id, _id):
     return result
 
 
+'''MAIL SENDER FUNCTIONS'''
+
+
+def randomword(length):
+    """
+    Generate a random string of given length
+    """
+    letters = string.ascii_lowercase
+    return ''.join(random.choice(letters) for i in range(length))
+
+
+def header(s):
+    if not s:
+        return None
+    s_no_newlines = ' '.join(s.splitlines())
+    return Header(s_no_newlines, 'utf-8')
+
+
+def template_params(paramsStr):
+    """
+    Translate the template params if they exist from the context
+    """
+    actualParams = {}
+    if paramsStr:
+        try:
+            params = json.loads(paramsStr)
+        except ValueError as e:
+            return_error('Unable to parse templateParams: {}'.format(str(e)))
+        # Build a simple key/value
+        for p in params:
+            if params[p].get('value'):
+                actualParams[p] = params[p]['value']
+            elif params[p].get('key'):
+                actualParams[p] = demisto.dt(demisto.context(), params[p]['key'])
+        return actualParams
+    else:
+        return None
+
+
+def transient_attachments(transientFile, transientFileContent, transientFileCID):
+    if transientFile is None or len(transientFile) == 0:
+        return []
+    attachments = []
+    entry_number = 0
+    for file_name in transientFile:
+        content_type, encoding = mimetypes.guess_type(file_name)
+        if content_type is None or encoding is not None:
+            content_type = 'application/octet-stream'
+
+        main_type, sub_type = content_type.split('/', 1)
+
+        if transientFileContent is not None and len(transientFileContent) > entry_number:
+            file_data = transientFileContent[entry_number]
+
+        else:
+            file_data = ''
+
+        if transientFileCID is not None and len(transientFileCID) > entry_number:
+            file_cid = transientFileCID[entry_number]
+
+        else:
+            file_cid = ''
+
+        attachments.append({
+            'name': file_name,
+            'maintype': main_type,
+            'subtype': sub_type,
+            'data': file_data,
+            'cid': file_cid
+        })
+        entry_number += 1
+
+    return attachments
+
+
+def handle_html(htmlBody):
+    """
+    Extract all data-url content from within the html and return as separate attachments.
+    Due to security implications, we support only images here
+    We might not have Beautiful Soup so just do regex search
```

Comment:
?

Reply:
Taken straight from MailSender - It seems to extract attachments which are embedded in the htmlBody.